process/unsigned/feeTxHandler.go

```diff
@@ -0,0 +1,236 @@
+	calculatedFeeTxs := ftxh.CreateAllUTxs()
+
+	miniBlocks := make(map[uint32]*block.MiniBlock)
+	for _, value := range calculatedFeeTxs {
+		dstShId := ftxh.address.ShardIdForAddress(value.GetRecvAddress())
+
+		txHash, err := core.CalculateHash(ftxh.marshalizer, ftxh.hasher, value)
+		if err != nil {
+			log.Debug(err.Error())
+			continue
+		}
+
+		var ok bool
+		var mb *block.MiniBlock
+		if mb, ok = miniBlocks[dstShId]; !ok {
+			mb = &block.MiniBlock{
+				ReceiverShardID: dstShId,
+			}
+		}
+
+		mb.TxHashes = append(mb.TxHashes, txHash)
+		miniBlocks[dstShId] = mb
+	}
+
+	return miniBlocks
+}
+
+// VerifyInterMiniBlocks verifies if transaction fees were correctly handled for the block
+func (ftxh *feeTxHandler) VerifyInterMiniBlocks(body block.Body) error {
+	err := ftxh.VerifyCreatedUTxs()
+	ftxh.CleanProcessedUTxs()
+
+	return err
+}
+
+// CleanProcessedUTxs deletes the cached data
+func (ftxh *feeTxHandler) CleanProcessedUTxs() {
+	ftxh.mutTxs.Lock()
+	ftxh.feeTxs = make([]*feeTx.FeeTx, 0)
+	ftxh.feeTxsFromBlock = make(map[string]*feeTx.FeeTx)
+	ftxh.mutTxs.Unlock()
+}
+
+func (ftxh *feeTxHandler) AddTxFeeFromBlock(tx data.TransactionHandler) {
+	currFeeTx, ok := tx.(*feeTx.FeeTx)
+	if !ok {
+		log.Debug(process.ErrWrongTypeAssertion.Error())
+		return
+	}
+
+	ftxh.mutTxs.Lock()
+	ftxh.feeTxsFromBlock[string(tx.GetRecvAddress())] = currFeeTx
+	ftxh.mutTxs.Unlock()
+}
+
+// AddProcessedUTx adds a new feeTx to the cache
+func (ftxh *feeTxHandler) AddProcessedUTx(tx data.TransactionHandler) {
+	currFeeTx, ok := tx.(*feeTx.FeeTx)
+	if !ok {
+		log.Debug(process.ErrWrongTypeAssertion.Error())
+		return
+	}
+
+	ftxh.mutTxs.Lock()
+	ftxh.feeTxs = append(ftxh.feeTxs, currFeeTx)
+	ftxh.mutTxs.Unlock()
+}
+
+func getPercentageOfValue(value *big.Int, percentage float64) *big.Int {
+	x := new(big.Float).SetInt(value)
+	y := big.NewFloat(percentage)
+
+	z := new(big.Float).Mul(x, y)
+
+	op := big.NewInt(0)
+	result, _ := z.Int(op)
+
+	return result
+}
+
+func (ftxh *feeTxHandler) createLeaderTx(totalGathered *big.Int) *feeTx.FeeTx {
+	currTx := &feeTx.FeeTx{}
+
+	currTx.Value = getPercentageOfValue(totalGathered, leaderPercentage)
+	currTx.RcvAddr = ftxh.address.LeaderAddress()
+
+	return currTx
+}
+
+func (ftxh *feeTxHandler) createBurnTx(totalGathered *big.Int) *feeTx.FeeTx {
+	currTx := &feeTx.FeeTx{}
+
+	currTx.Value = getPercentageOfValue(totalGathered, burnPercentage)
+	currTx.RcvAddr = ftxh.address.BurnAddress()
+
+	return currTx
+}
+
+func (ftxh *feeTxHandler) createCommunityTx(totalGathered *big.Int) *feeTx.FeeTx {
+	currTx := &feeTx.FeeTx{}
+
+	currTx.Value = getPercentageOfValue(totalGathered, communityPercentage)
+	currTx.RcvAddr = ftxh.address.ElrondCommunityAddress()
+
+	return currTx
+}
+
+// CreateAllUtxs creates all the needed fee transactions
+// According to economic paper 50% burn, 40% to the leader, 10% to Elrond community fund
+func (ftxh *feeTxHandler) CreateAllUTxs() []data.TransactionHandler {
```

Comment:
Typo in comment, should start with: `CreateAllUTxs`

Reply:
done